### ISSUE TITLE
Blazor Loader/Publisher Improvements

### ISSUE THREAD

**Issue by elepner** [dotnet/aspnetcore#51821]
## Summary

Blazor Wasm has a large bundle size and there's no way to avoid this. However there are steps to improve UX during loading. We use Blazor in production for thousands of our customers and quite happy with it but we had to implement quite complex loader by ourselves which solves the issues I want to describe this design proposal. 

## Motivation and goals

Very first application experience must go for a user as smooth as possible, thus Blazor loader should provide better capabilities than it does right now.

## In scope

Here's the list that we had to implement in our DIY loader (it took us substantial amount of efforts in development and what's more important in testing. It'd be nice if those features would be available for everyone out of the box.
1. Downloading progress in bytes should be available ( #46549). As of now blazor provides % of downloaded bundle. It just counts files to be downloaded vs downloading ignoring that >50% of the bundle is dotnet.wasm file. Thus progress would stuck on the last step for a long time pissing off users badly.
1. Better CDN integration:
     1. Append hashes to file names in the bundle (#47575 ). Replacing files on CDN is quite bad idea because of heavy caching. We have experienced crashes of the application i production because wrong old version of some DLL was loaded. Blazor already computes hashes and exposes them through blazor.boot.json. Why not giving an option to append them to the file name? IMO, it should be default option like JS frameworks do.
     3. Give an ability to control maximum individual file size in the bundle to enable compression CDNs (#51772)

## In conclusion

Since Blazor loader allows for any kind of customization it let us implement our requirements which is good. However, there's nothing extraordinary in those demands. Every Blazor developer would come across one of them eventually, so it's better that the platform would provide such opportunities out of the box.

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us and sharing your scenarios, @elepner.
Given all these asks are already tracked by unique issues, we're going to close this one. Please upvote the linked issues (those you haven't filed yourself) so that we take your feedback into consideration when handling .NET 9 prioritization.
